app/javascript/packs/search.js

```diff
@@ -27,32 +43,84 @@ $( function () {
   });
 
   //Add badge for date filters if populated
-  if (parseInt($("#starting_year").val()) || parseInt($("#ending_year").val())) {
+  if (initialStartDate || initialEndDate) {
     publicationDateValidation();
   }
 
   // Listener for checkboxes
+  let filtersOnDeck = new Set();
   $(".sidebar input:checkbox").on('change', function(){
     const parentId = $(this).parent().parent().parent()[0].id; // "sections" or "agencies"
+    showActiveFilters();
+
+    if (filtersOnDeck.has(this.id)) {
+      filtersOnDeck.delete(this.id);
+    }
+    else {
+      filtersOnDeck.add(this.id);
+    }
+
     if (this.checked) {
       addBadge(this.id, parentId);
-    } else {
+    }
+    else {
       removeBadge(this.id, parentId);
     }
+
+    if (filtersOnDeck.size > 0) {
+      checkboxChangeFlag = true;
+    }
+    else {
+      checkboxChangeFlag = false;
+    }
+    setApplyButton();
   });
 
   // Validate the publication date input
-  $("#starting_year").on("change", publicationDateValidation);
-  $("#ending_year").on("change", publicationDateValidation);
+  $("#starting_year").on("change", detectDateChange);
+  $("#ending_year").on("change", detectDateChange);
+
+  // Detects if the change to publication date range is different from load
+  // and displays the apply button if so
+  function detectDateChange() {
+    let currentStartDate = parseInt($("#starting_year").val());
+    let currentEndDate = parseInt($("#ending_year").val());
+    if(publicationDateValidation()) {
+      if (isNaN(initialStartDate) && isNaN(currentStartDate) && isNaN(initialEndDate) && isNaN(currentEndDate)) {
+        dateChangeFlag = false;
+      }
+      else if (currentStartDate === initialStartDate && currentEndDate === initialEndDate) {
+        dateChangeFlag = false;
+      }
+      else {
```

Comment:
to DRY this up you could initialize the value as false and then only make a single conditional that sets the value to `true`